cloud-canary.py

```diff
@@ -99,136 +100,154 @@ def main():
     return args
 
 
-def deploy_instance(args):
-    api_key = args['acskey']
-    secret_key = args['acssecret']
-    zonename = args['zonename']
-    endpoint = "https://" + args['endpoint'] + "/compute"
-    template = args['template']
-    offering = args['offering']
+def create_network(cs, name, zoneid):
+    """Creates a managed private network"""
 
-    logging.basicConfig(
-        format='%(asctime)s %(pathname)s %(levelname)s:%(message)s',
-        level=logging.DEBUG,
-        filename=logfile)
-    logging.getLogger().addHandler(logging.StreamHandler())
+    noid = cs.listNetworkOfferings()['networkoffering'][0]['id']
+    return cs.createNetwork(
+            displaytext="cloud-canary test network",
+            name=name, networkofferingid=noid,
+            zoneid=zoneid,
+            startip="10.0.0.1",
+            endip="10.0.0.100",
+            netmask="255.255.255.0",
+            )['network']
 
-    cs = CloudStack(endpoint=endpoint,
-                    key=api_key,
-                    secret=secret_key,
-                    timeout=60)
 
-    location = [location for location in cs.listZones()['zone']
-                if location['name'].lower() == zonename.lower()][0]
+def deploy_instance(cs, name, template, zoneid, offering, networkid):
+    """Deploys an instance"""
 
-    logging.info("Zone selected : %s", location)
+    try:
+        for node in cs.listVirtualMachines()['virtualmachine']:
```

Comment:
```suggestion
        for node in cs.listVirtualMachines(name=name, fetch_list=True):
```

* This removes the need for the `try… except KeyError` dance.
* filtering by name can be done at the API level